extensions/s3/src/main/java/io/deephaven/extensions/s3/S3ChannelContext.java

```diff
@@ -63,27 +66,43 @@ final class S3ChannelContext implements SeekableChannelContext {
         }
     }
 
-    public long size(S3Uri uri) throws IOException {
-        assume(uri);
-        populateSize();
+    void verifyOrSetUri(S3Uri uri) {
+        if (this.uri == null) {
+            this.uri = Objects.requireNonNull(uri);
+        } else if (!this.uri.equals(uri)) {
+            throw new IllegalStateException(
+                    String.format("Inconsistent URIs. expected=%s, actual=%s", this.uri, uri));
```

Comment:
Should our exceptions include `ctxStr()`?

Reply:
Sure